ios/Sources/GutenbergKit/Sources/Views/Patterns/PatternListView.swift

```diff
@@ -0,0 +1,46 @@
+import SwiftUI
+
+/// A full-screen view showing all patterns in a category
+struct PatternListView: View {
+    let section: PatternSection
+    let onPatternSelected: (Pattern) -> Void
+    let viewportWidth: Int?
+
+    @Environment(\.dismiss) private var dismiss
+
+    init(section: PatternSection, onPatternSelected: @escaping (Pattern) -> Void, viewportWidth: Int? = nil) {
+        self.section = section
+        self.onPatternSelected = onPatternSelected
+        self.viewportWidth = viewportWidth
+    }
+
+    var body: some View {
```

Comment:
Should we consider adding the viewport toggle to a header for this view? It could be useful to toggle the viewport while viewing a list of patterns in a category.